src/ethereum_spec_tools/sync.py

```diff
@@ -77,19 +81,39 @@ def module(self, name: str) -> Any:
         """
         return self.active_fork.module(name)
 
-    def set_block(self, block_number: int) -> None:
+    def set_block(self, block_number: int, block_timestamp: int) -> None:
         """Set the block number and switch to the correct fork."""
         self.block_number = block_number
         self.active_fork_index = 0
-        while self.next_fork and block_number >= self.next_fork.block:
+        while (
+            self.next_fork
+            and hasattr(self.next_fork, "block")
+            and block_number >= self.next_fork.block
```

Comment:
I'm really leaning towards the OOP approach we discussed on the call a while back, where we have `Timestamp` and `BlockNumber` classes that each know how to test a block to see if it's time.